test/goldens/end2end/sdp_transform-0.2.0.json

```diff
@@ -108,5 +94,6 @@
     "branch": "master",
     "isVerified": true
   },
-  "urlProblems": []
+  "urlProblems": [],
+  "errorMessage": "`dart pub get` failed:\n\n```\nOUT:\nResolving dependencies...\nERR:\npubspec.yaml has no lower-bound SDK constraint.\nYou should edit pubspec.yaml to contain an SDK constraint:\n\nenvironment:\n  sdk: '>=2.10.0 <3.0.0'\n\nSee https://dart.dev/go/sdk-constraint\n```"
```

Comment:
Any idea why this has changed? It looks like something that shouldn't change if we keep the same SDK version (2.16.1 on GitHub Actions).

Reply:
There is a distinction between whether or not we consume a package directly as root package, or as a dependency. Dependencies don't have to specify SDK constraints, but instead get an inferred constraint.

This is because some old packages were published without SDK constraints.

@sigurdm, this might be something we should do something about.. any ideas?

Reply:
Yeah not sure what is best here...

We could add the inferred constraint before running `pub outdated` if we need...

Reply:
Or we create a dummy package in a temporary directory with a dependency on this package.
But then we have to depend on it from pub.dev, not using a path-dependency, because that could cause conflicts with cycles.

Reply:
> Or we create a dummy package in a temporary directory with a dependency on this package.

I guess that is what we already do?

In that case it seems we are living with the cycle case...

Reply:
If we depend on it locally we need to rewrite the pubspec.yaml, that's also fine...

Reply:
Ah - no we do this:
```
  /// Removes the dev_dependencies from the pubspec.yaml
  /// Adds lower-bound minimal SDK constraint - if missing.
  /// Returns the backup file with the original content.
  Future<File> _stripAndAugmentPubspecYaml(String packageDir) async {
  ```
  
  I guess if we already are doing this we could just continue doing it (though it is ugly as muck).